tests/test_standalone_stac_item.py

```diff
@@ -0,0 +1,30 @@
+import json
+
+import requests
+import xncml
+
+from STACpopulator.implementations.CMIP6_UofT.add_CMIP6 import (
+    CMIP6ItemProperties,
+    make_cmip6_item_id,
+)
+from STACpopulator.models import GeoJSONPolygon
+from STACpopulator.stac_utils import STAC_item_from_metadata
+
+
+def test_standalone_stac_item():
+    url = (
+        "https://pavics.ouranos.ca/twitcher/ows/proxy/"
+        "thredds/ncml/birdhouse/testdata/xclim/cmip6/sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc"
+        "?catalog=https%3A%2F%2Fpavics.ouranos.ca%2Ftwitcher%2Fows%2Fproxy%2F"
+        "thredds%2Fcatalog%2Fbirdhouse%2Ftestdata%2Fxclim%2Fcmip6%2Fcatalog.html"
+        "&dataset=birdhouse%2Ftestdata%2Fxclim%2Fcmip6%2Fsic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc"
+    )
+
+    attrs = xncml.Dataset.from_text(requests.get(url).content).to_cf_dict()
+    stac_item_id = make_cmip6_item_id(attrs["attributes"])
+    stac_item = STAC_item_from_metadata(stac_item_id, attrs, CMIP6ItemProperties, GeoJSONPolygon)
```

Comment:
There is still a strong relationship between those components. 
`STAC_item_from_metadata` can only parse attributes if they respect exactly the format returned by `xncml.Datase.to_cf_dict` (i.e.: `cfmeta = attrs["groups"]["CFMetadata"]["attributes"]`). 

The same applies for `make_cmip6_item_id` that expect a very specific set of attributes that come from `xncml.Datase.to_cf_dict`.

All this block should be a single `CMIP6populator.get_stac_item` method that takes the URL, and does everything necessary to obtain the STAC Item without any futher intervention. 

Both `make_cmip6_item_id` and `STAC_item_from_metadata` should be converted to methods under `CMIP6populator`. There is no reason for them to exist elsewhere given how they heavily depend on the specific formats that `CMIP6populator.item_properties_model` and `CMIP6populator.item_geometry_model` defines.

Reply:
> Both `make_cmip6_item_id` and `STAC_item_from_metadata` should be converted to methods under `CMIP6populator`. There is no reason for them to exist elsewhere given how they heavily depend on the specific formats that `CMIP6populator.item_properties_model` and `CMIP6populator.item_geometry_model` defines.

Interesting suggestion... I can see what you mean. `make_cmip6_item_id` is easy, but `STAC_item_from_metadata` is tricky. I don't necessarily want to move most of that logic into `CMIP6populator` as there is a lot there that is common to making any STAC item. But maybe one will have to.... Let me chew on this. It might be that we keep this for a later.

Reply:
I agree. Some part of the logic should be under `STACPopulatorBase.stac_item_from_metadata`, since it applies to any STAC Item generation case (for example, handling the extents, date-time, and other generic metadata).
The `CMIP6populator.stac_item_from_metadata` specificatlly for converting CF attributes to STAC CMIP6 extension would then do something like: 

```python 
class CMIP6populator(STACPopulatorBase):
    def stac_item_from_metadata(self, item_data):
        stac_base = super().stac_item_from_metadata(item_data)
        stac_cmip6 = # specific populator logic for CF...
        stac_item = # merge stac_base + stac_cmip6
        return stac_item
```